### PR TITLE
feat: disable minio for CN >= 0.74.0

### LINKED ISSUE #4166: default the consensus node to use block streams only and not both when using TSS (CN v0.74+) and disable minio
this is required to properly benchmark the creation, confirmation, and verification of a transactions round trip time for CN > BN > MN 

a more detailed explanation for the reasoning can be found in https://github.com/hiero-ledger/roadmap/issues/42

Additionally, disable the minio operator and minio pool as it is only needed for RECORDS/BOTH, so disabling it will cause a smaller memory footprint.  NOTE: disabling minio is a breaking change for some unique/rare scenarios.  we should let the BN team know about it.  We should be able to just default the solo minio flag to false.

- required by https://github.com/hiero-ledger/solo/issues/4167

### PR DESCRIPTION
**Pull request by JeffreyDallas**
## Description

This pull request changes the following:

* disables MinIO for CN `>= v0.74.0` when block-node integration is active
* keeps MinIO/record-stream behavior for CN `>= v0.74.0` when no block node is deployed
* preserves older block stream settings during upgrade flows instead of forcing a new default during later maintenance steps
* updates mirror-node block-node integration values so importer no longer tries to download record/balance streams from MinIO in block-node mode
* updates one-shot/example/CI coverage so scenarios that should stay on MinIO can explicitly do so, and scenarios that should exercise block-node mode do so consistently
* adds supporting robustness fixes uncovered while validating this change across standard, dual-cluster, migration, node-update, relay, and external-database flows

### Runtime behavior

For consensus node versions before `v0.74.0`:

* existing behavior is preserved
* block stream mode remains the legacy default (`BOTH`)
* MinIO-backed record stream flow is unchanged

For consensus node versions `>= v0.74.0`:

* if a block node is deployed/configured for the network, Solo now:
  * sets `blockStream.streamMode` to `BLOCKS` by default
  * disables `cloud.minio.enabled`
  * disables MinIO-related uploader sidecars
  * keeps mirror importer on the block-node path instead of record/balance downloaders
* if no block node is deployed/configured, Solo now:
  * keeps MinIO enabled
  * keeps record-stream based behavior
  * uses `RECORDS` as the effective stream mode for new deployments on CN `>= v0.74.0`

Upgrade behavior:

* when an older network is upgraded, Solo preserves already-valid existing stream mode settings (`BOTH` or `BLOCKS`) instead of silently rewriting them during later block-node/profile maintenance steps
* this is important for version-upgrade and migration scenarios that start on older CN versions and later move to `>= v0.74.0`

#### Network deployment

* determine whether MinIO should be enabled from the combination of:
  * consensus node version
  * whether block node integration is actually configured
* only prepare MinIO secrets when MinIO is enabled
* still create the stream uploader secret when required by downstream components
* write chart overrides to disable:
  * `cloud.minio.enabled`
  * `cloud.generateNewSecrets`
  * record/event/blockstream uploader sidecars
  when block-node mode is active on CN `>= v0.74.0`
* skip MinIO readiness checks when MinIO is intentionally disabled

#### Block node and application.properties handling

* centralize block-stream-mode resolution by CN version and block-node presence
* update `block-nodes.json` copy logic to also reconcile `application.properties` stream mode/writer mode when block-node configuration is pushed to consensus nodes
* make block-node source generation skip the current block node and compact source indexes correctly
* make block-node config copy more resilient by waiting for a ready pod and avoiding stale pod references

#### Mirror node

* when block-node integration is enabled, explicitly disable mirror importer record and balance downloaders so mirror no longer tries to fetch streams from MinIO

#### One-shot flows

* one-shot block-node deployment now honors `ONE_SHOT_WITH_BLOCK_NODE` consistently through:
  * remote config component creation
  * block-node deployment
  * destroy flow
  * cluster setup argv generation
* when one-shot block-node mode is enabled, cluster setup passes `--no-minio`

#### Supporting fixes discovered during validation

These were needed to keep existing validation flows passing while changing the storage/streaming behavior:

* relay network map generation now prefers HAProxy load balancer IPs when available, which fixes relay-to-consensus connectivity in multi-cluster layouts
* pod selection logic now avoids stale/terminal pods and prefers pods with valid references
* profile/application-properties staging logic re-applies Solo-managed values after merging user properties so required deployment settings are not lost

### Related Issues

* Closes #4166

### Pull request (PR) checklist

* [x] This PR added tests (unit, integration, and/or end-to-end)
* [x] This PR updated documentation
* [ ] This PR added no TODOs or commented out code
* [x] This PR has no breaking changes
* [ ] Any technical debt has been documented as a separate issue and linked to this PR
* [x] Any `package.json` changes have been explained to and approved by a repository manager
* [x] All related issues have been linked to this PR
* [x] All changes in this PR are included in the description
* [x] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* [x] This PR added unit tests
* [x] This PR added integration/end-to-end tests
* [x] These changes required manual testing that is documented below
* [x] Anything not tested is documented

The following manual testing was done:

* all PR checks passed for this branch
* all example tests in the examples workflow passed for this branch
* validated one-shot block-node flow with MinIO disabled
* validated non-block-node CN `>= v0.74.0` flow keeps MinIO enabled
* validated migration / version-upgrade handling of pre-`v0.74.0` stream-mode settings
* validated external-database and relay flows after mirror importer/block-node changes
* validated local-build-path example behavior for node-update-transaction

The following was not tested:

* no additional manual scenarios beyond the branch CI/examples coverage listed above

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: .github/workflows/flow-performance-test.yaml [hunk 1]
```diff
@@ -162,6 +162,9 @@ jobs:
           fi
           echo MIRROR_NODE_PINGER_TPS=10 >> .env
           echo ONE_SHOT_METRICS_TEST_DURATION_IN_MINUTES=5 >> .env
+          # Workaround: keep BOTH mode so SmartContractLoadTest does not return INVALID_TRANSACTION_BODY.
```
Comment (jan-milenkov) [inline .github/workflows/flow-performance-test.yaml:165]: Create relevant issue if not already done, link it to the PR/Original Issue and mark it as blocked, I think P1 priority is appropriate

File: examples/hardhat-with-solo/Taskfile.yml [hunk 2]
```diff
@@ -37,7 +37,7 @@ tasks:
     desc: Destroy the Solo network using one-shot single destroy
     cmds:
       # Solo one-shot single destroy
-      - cmd: $SOLO_COMMAND one-shot single destroy
+      - cmd: ONE_SHOT_WITH_BLOCK_NODE="${ONE_SHOT_WITH_BLOCK_NODE:-true}" $SOLO_COMMAND one-shot single destroy
```
Comment (jan-milenkov) [inline examples/hardhat-with-solo/Taskfile.yml]: Optional change but recommended,

You  can pass the env via a `env`/`envs` block, this will make the command invocations a lot cleaner

File: .github/workflows/flow-performance-test.yaml (hunk 1 above)
Comment (JeffreyDallas) [reply .github/workflows/flow-performance-test.yaml:165]: Isse https://github.com/hiero-ledger/solo/issues/4708 created

File: examples/hardhat-with-solo/Taskfile.yml (hunk 2 above)
Comment (JeffreyDallas) [reply examples/hardhat-with-solo/Taskfile.yml]: changed to task-level env

**Review by jan-milenkov — DISMISSED** [review]

**Review by jeromy-cannon — APPROVED** [review]
